modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -213,27 +215,28 @@ package object engine {
     * It also updates the `State` as needed.
     */
   private def execute(id: Sequence.Id): HandleP[Unit] = {
+
     // Send the expected event when the `Action` is executed
     // It doesn't catch run time exceptions. If desired, the Action as to do it itself.
-    def act(t: (Action, Int)): Process[Task, Event] = t match {
+    def act(t: (Action, Int), cx: ActionMetadata): Process[Task, Event] = t match {
       case (action, i) =>
-        Process.eval(action).flatMap {
+        Process.eval(action(cx)).flatMap {
           case r@Result.OK(_)         => Process(completed(id, i, r))
-          case r@Result.Partial(_, c) => Process(partial(id, i, r)) ++ act((c, i))
+          case r@Result.Partial(_, c) => Process(partial(id, i, r)) ++ act((c, i), cx)
           case e@Result.Error(_)      => Process(failed(id, i, e))
         }
     }
 
-    getS(id).flatMap(
-      _.map { seq =>
+    get.flatMap(st => st.sequences.get(id).map { seq =>
         seq match {
           case Sequence.State.Final(_, _) =>
             // The sequence is marked as completed here
             putS(id)(seq) *> send(finished(id))
-          case _                          =>
-            val u = seq.current.actions.zipWithIndex.map(act)
+          case _                          => {
+            val u = seq.current.actions.zipWithIndex.map(x => act(x, ActionMetadata(st.conditions, st.operator, seq.toSequence.metadata.observer)))
```

Comment:
I can see that this is only set when the current `Execution` actually being executed, so what's in the `ActionMetadata` before, in the pending Executions when the Sequence is loaded?